orders/src/main/java/com/orders/crudyorders/model/Customer.java

```diff
@@ -0,0 +1,137 @@
+package com.orders.crudyorders.model;
+
+import javax.persistence.*;
+
+@Entity
+@Table(name = "customer")
+public class Customer
+{
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(nullable = false)
+    private long custcode;
+
+    @Column(nullable = false)
+    private String custname;
+
+    private String custcity, workingarea, custcountry, grade;
+    private double openingamt, recieveamt, paymentamt, outstandingamt;
+    private String phone;
+
+    @OneToMany
+    @JoinColumn(name = "agentcode", nullable = false)
+    // connects this to the agent model
+    private long agentcode;
```

Comment:
• Set this as a many-to-many relationship and the name for the `@JoinColumn` annotation must be the same as the ID field of the type you are referencing (`agentcode` here)
• Foreign keys aren't done with ID data types like `long` here; just directly use the class of the object you are referencing (in this case, `Agent`)
• Using the `@JsonIgnoreProperties` annotation, prevent the `customers` and `orders` properties of your `Agent` object from being displayed (or else the customer's agent will display the customer again, and so forth in an infinite loop) - this will be sent as an array of strings with the notation `({"property1", "property2"})`
• Don't forget to include this property in your constructor, getters and setters

The directions above should result in:
```
@ManyToOne
@JoinColumn(name = "agentcode", nullable = false)
@JsonIgnoreProperties({"customers", "orders"})
private Agent agent;
```